v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/SourceWriterFn.java

```diff
@@ -188,10 +182,15 @@ public void processElement(ProcessContext c) {
                             > Long.parseLong(spannerRec.getRecordSequence())));
 
         if (!isSourceAhead) {
-          MySqlDao mySqlDao = mySqlDaoMap.get(shardId);
+          IDao sourceDao = sourceProcessor.getSourceDaoMap().get(shardId);
```

Comment:
Just have a getDao method that takes the shardId and give the Dao as response, we don't need to expose that SourceProcessor internally uses a map